scripts/generate-credentials-with-sts-assume-role.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# This script uses aws sts assume-role to generate temporary credentials
+# and outputs them in $GITHUB_OUTPUT so those can be used in other workflow jobs.
+# role-arn = arn:aws:iam::358363220050:role/terraform-provider-mongodbatlas-acceptancetests
+
+# Define a function to convert a string to lowercase
+function to_lowercase() {
+  echo "$1" | tr '[:upper:]' '[:lower:]'
+}
+# Convert the input string to lowercase
+aws_region=$(to_lowercase "$AWS_REGION")
+# Replace all underscores with hyphens
+aws_region=${aws_region//_/-}
+# e.g. from US_EAST_1 to us-east-1
+
+# Get the STS credentials
+export AWS_REGION="$aws_region"
+CREDENTIALS=$(aws sts assume-role --role-arn "$ASSUME_ROLE_ARN" --role-session-name newSession --output text --query 'Credentials.[AccessKeyId, SecretAccessKey, SessionToken]')
```

Comment:
do we want to check if aws command fails?

Reply:
going to add `set -Eeou pipefail` at the start of the script. How does that sound?

Reply:
that's great! we should probably use that in all/most scripts.